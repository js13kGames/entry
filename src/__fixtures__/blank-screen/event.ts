import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 1,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/blank-screen",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "360b838";
