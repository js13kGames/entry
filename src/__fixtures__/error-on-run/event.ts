import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 6,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/error-on-run",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
