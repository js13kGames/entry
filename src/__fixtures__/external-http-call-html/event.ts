import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 5,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/external-http-call-html",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "c9a43c6";
