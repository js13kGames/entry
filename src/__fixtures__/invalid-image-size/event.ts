import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 12,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/invalid-image-size",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "ce740ea";
