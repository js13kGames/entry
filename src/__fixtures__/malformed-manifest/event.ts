import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 10,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/malformed-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
