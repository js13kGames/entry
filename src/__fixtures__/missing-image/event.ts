import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 11,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-image",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "400d963";
