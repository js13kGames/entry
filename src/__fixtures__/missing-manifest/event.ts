import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 9,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "de7f563";
