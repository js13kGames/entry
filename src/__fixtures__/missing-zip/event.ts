import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 3,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-zip",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "e60be98";
