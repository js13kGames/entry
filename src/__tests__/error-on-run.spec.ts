import * as event from "../__fixtures__/error-on-run/event";
import { bootstrap } from "./_bootstrap";

describe("integration  error-on-run", () => {
  bootstrap(event);
});
