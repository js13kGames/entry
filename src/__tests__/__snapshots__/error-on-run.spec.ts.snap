// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration  error-on-run should analyze 1`] = `
Array [
  Object {
    "assetFiles": Array [
      "bundle.zip",
    ],
    "conclusion": "success",
    "name": "bundle-found",
  },
  Object {
    "conclusion": "success",
    "name": "bundle-unziped",
  },
  Object {
    "bundleSize": 3882,
    "conclusion": "success",
    "name": "bundle-size",
    "sizeLimit": 13312,
  },
  Object {
    "bundleFiles": Array [
      "asset/text.txt",
      "asset/script.js",
      "asset/image/1659821.jpeg",
      "index.html",
    ],
    "conclusion": "success",
    "deployUrl": "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/index.html",
    "name": "index-found",
  },
  Object {
    "conclusion": "failure",
    "errors": Array [
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/script.js 7:10 Uncaught Error: something bad",
    ],
    "name": "run-without-error",
  },
  Object {
    "conclusion": "success",
    "externalUrls": Array [],
    "name": "run-without-external-http",
    "urls": Array [
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/index.html",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/image/1659821.jpeg",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/script.js",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/favicon.ico",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/text.txt",
    ],
  },
  Object {
    "conclusion": "success",
    "name": "run-without-blank-screen",
    "screenShotUrl": "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/game_preview_screenshot.png",
  },
]
`;

exports[`integration  error-on-run should generate checks 1`] = `
Array [
  Object {
    "conclusion": "success",
    "name": "bundle found",
    "output": Object {
      "summary": "",
      "text": "> bundle.zip",
      "title": "",
    },
  },
  Object {
    "conclusion": "success",
    "name": "bundle unziped",
    "output": Object {
      "summary": "",
      "title": "",
    },
  },
  Object {
    "conclusion": "success",
    "name": "bundle size",
    "output": Object {
      "summary": "",
      "text": "> 3882, 13312",
      "title": "",
    },
  },
  Object {
    "conclusion": "success",
    "name": "index found",
    "output": Object {
      "summary": "",
      "text": "> http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/index.html
> asset/text.txt, asset/script.js, asset/image/1659821.jpeg, index.html",
      "title": "",
    },
  },
  Object {
    "conclusion": "failure",
    "name": "run without error",
    "output": Object {
      "summary": "",
      "text": "> - http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/script.js 7:10 Uncaught Error: something bad",
      "title": "",
    },
  },
  Object {
    "conclusion": "success",
    "name": "run without external http",
    "output": Object {
      "summary": "",
      "text": "> http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/index.html, http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/image/1659821.jpeg, http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/script.js, http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/favicon.ico, http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/text.txt
> ",
      "title": "",
    },
  },
  Object {
    "conclusion": "success",
    "name": "run without blank screen",
    "output": Object {
      "summary": "",
      "text": "> http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/game_preview_screenshot.png",
      "title": "",
    },
  },
]
`;

exports[`integration  error-on-run should generate report 1`] = `
"Hello ðŸ‘‹
I am a bot. My job is to review your submission.

I reviewed your latest release, [entry-error-on-run-1](https://github.com/platane-org/js13-entry/releases/tag/entry-error-on-run-1)

Your game seems to have error at launch with chrome
I got this error:
\`\`\`
http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/asset/script.js 7:10 Uncaught Error: something bad
\`\`\`
Can you take a look ? I have it deployed [here](http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/b3107c5e2f9240efdff74a17a6e5ef0b/index.html)"
`;
